app/controllers/api/status_controller.rb

```diff
@@ -5,32 +5,26 @@ def check
     if params[:projects].any?
       # Try to get all the projects passed in.
       @projects = params[:projects].group_by{|project| project[:platform] }.map do |platform, projects|
-        projects.each_slice(1000).map do |slice|
-          Project.platform(platform).where(name: slice.map{|project| project[:name] }.uniq).includes(:repository, :versions, :repository_maintenance_stats)
-        end
+        projects.each_slice(1000).map{|slice| find_projects(slice, platform)}
```

Comment:
Love breaking this out into another method!

Style-wise, I'd consider re-writing it as something like:
```ruby
@projects = params[:projects]
  .group_by { |project| project[:platform] }
  .flat_map(&method(:find_projects)) # moving the slicing into find_projects
  .compact
```